### DISCUSSION TITLE
Migrating FreeMarker account forms from Keycloak 20 to 21

### DISCUSSION THREAD

**Discussion by ssitzo1** [keycloak/keycloak#22368] [Q&A] [answered]
Hi everyone,

we are finally updating Keycloak from version 20 to 21 and migrating providers.

We have quite a few themes that extend the base theme and in some of them, we also use custom providers for email and account to add additional data/sub pages to templates.

I had no problems migrating the email providers (they extend [FreeMarkerEmailTemplateProvider](https://github.com/keycloak/keycloak/blob/20.0.5/services/src/main/java/org/keycloak/email/freemarker/FreeMarkerEmailTemplateProvider.java)), but I'm having a hard time migrating custom account providers which extend the [FreeMarkerAccountProvider](https://github.com/keycloak/keycloak/blob/20.0.5/services/src/main/java/org/keycloak/forms/account/freemarker/FreeMarkerAccountProvider.java). Package `org.keycloak.forms.account.freemarker` is no longer available in version 21 and I cannot find any migration steps on this topic.

Is there something I have missed?

**Comment by ssilvert** [comment, marked as answer]
We have dropped support for account console V1.  However, there is this
solution that should be ready soon:
https://github.com/keycloak/keycloak/issues/22318
https://github.com/keycloak/keycloak/pull/22317
